BossMod/Components/Towers.cs

```diff
@@ -46,6 +47,34 @@ public override void AddHints(int slot, Actor actor, TextHints hints)
         }
     }
 
+    public override void AddAIHints(int slot, Actor actor, PartyRolesConfig.Assignment assignment, AIHints hints)
+    {
+        // draw hints for imminent, currently unsoaked towers
+        // we consider some list of towers part of the same "group" if their activations are within 500ms, as there can be varying delays between helper actors in an encounter casting the "same" spell
+        // generally, successive towers that are meant to be soaked by one player (i.e. in quest battles) activate no more frequently than about 2 seconds apart, so this is pretty conservative
+        var imminentMissedTowers = Towers.Where(t => ShouldHint(slot, actor, t))
+            .GroupBy(x => x.Activation.Round(TimeSpan.FromMilliseconds(500)))
+            .MinBy(gt => gt.Key);
+
+        if (imminentMissedTowers != null)
```

Comment:
If I understand correctly, it could just be done in a way similar to other components:
```
var deadline = Towers[0].Activation.AddSeconds(0.5);
List<Func<WPos, float>> imminentTowers = [.. Towers.TakeWhile(t => t.Activation <= deadline).Select(t => ShapeDistance.InvertedCircle(t.Position, t.Radius)];
hints.AddForbiddenZone(ShapeDistance.Intersection(imminentTowers);
```